src/vmspawn/vmspawn-util.c

```diff
@@ -0,0 +1,341 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "architecture.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "json.h"
+#include "log.h"
+#include "macro-fundamental.h"
+#include "memory-util.h"
+#include "path-lookup.h"
+#include "path-util.h"
+#include "sort-util.h"
+#include "string-util-fundamental.h"
+#include "string-util.h"
+#include "recurse-dir.h"
+#include "strv.h"
+#include "vmspawn-util.h"
+#include <stdio.h>
+
+OvmfConfig* ovmf_config_new(void) {
+        return new0(OvmfConfig, 1);
+}
+
+OvmfConfig* ovmf_config_free(OvmfConfig *config) {
+        config->path = mfree(config->path);
+        config->vars = mfree(config->vars);
+        return mfree(config);
+}
+
+bool qemu_check_kvm_support(void) {
+        int r;
+        r = access("/dev/kvm", F_OK);
+        if (r == 0)
+                return true;
+
+        switch (errno) {
+        case ENOENT:
+                log_debug_errno(errno, "/dev/kvm not found. Not using KVM acceleration.");
+                break;
+
+        case EACCES:
+                log_debug_errno(errno, "Permission denied to access /dev/kvm. Not using KVM acceleration.");
+                break;
+        }
+
+        return false;
+}
+
+/* holds the information needed to sort the config files into the right order */
+typedef struct OvmfConfigFile {
+        char *name;
+        int priority;
+} OvmfConfigFile;
+
+static inline void ovmf_config_file_free_many(OvmfConfigFile *confs, size_t n_confs) {
+        assert(confs || n_confs == 0);
+
+        for (size_t i = 0; i < n_confs; i++) {
```

Comment:
`FOREACH_ARRAY` here. Also please remove the brackets.

Reply:
Will replace with FOREACH_ARRAY.

Reply:
Fixed in f8c2832218.